ftw/simplelayout/aliasblock/contents/aliasblock.py

```diff
@@ -0,0 +1,107 @@
+from ftw.referencewidget.selectable import DefaultSelectable
+from ftw.referencewidget.sources import ReferenceObjSourceBinder
+from ftw.referencewidget.widget import ReferenceBrowserWidget
+from ftw.simplelayout import _
+from ftw.simplelayout.aliasblock.contents.interfaces import IAliasBlock
+from plone import api
+from plone.uuid.interfaces import IUUID
+from plone.autoform.interfaces import IFormFieldProvider
+from plone.dexterity.content import Item
+from plone.directives import form
+from plone.directives.form import widget
+from z3c.form import validator
+from z3c.relationfield.schema import RelationChoice
+from zope.interface import alsoProvides
+from zope.interface import implements
+from zope.interface import Invalid
+from Products.CMFPlone.interfaces.siteroot import IPloneSiteRoot
+
+
+class AliasBlockSelectable(DefaultSelectable):
+
+    def is_selectable(self):
+        if not self.content:
+            return False
+
+        selectable = super(AliasBlockSelectable, self).is_selectable()
+        is_sl_page = self.content.portal_type == 'ftw.simplelayout.ContentPage'
+
+        # Don't allow sl pages containing another AliasBlock
+        if is_sl_page:
+            if (not IPloneSiteRoot.providedBy(self.source.context) and
+                IUUID(self.content) == IUUID(self.source.context)):
+                return False
+            return not bool(filter(
+                lambda item: item.portal_type == 'ftw.simplelayout.AliasBlock',
+                self.content.objectValues()
+            ))
+        return selectable
+
+
+def get_selectable_blocks():
+    return ['ftw.simplelayout.TextBlock',
+            'ftw.simplelayout.GalleryBlock',
+            'ftw.simplelayout.FileListingBlock',
+            'ftw.sliderblock.SliderBlock',
+            'ftw.news.NewsListingBlock',
+            'ftw.events.EventListingBlock',
+            'ftw.iframeblock.IFrameBlock',
+            'ftw.addressblock.AddressBlock',
+            'ftw.simplelayout.MapBlock',
+            'ftw.simplelayout.ContentPage']
+
+
+class IAliasBlockSchema(form.Schema):
+    """AliasBlock for simplelayout
+    """
+
+    widget('alias', ReferenceBrowserWidget,
+           start='parent',
+           )
+    alias = RelationChoice(
+        title=_(u'label_alias_content',
+                default=u'Alias Content'),
+        description=_(
+            u'label_alias_description',
+            default=u'Choose a block to be rendered within this block.'),
+        required=True,
+        source=ReferenceObjSourceBinder(
+            selectable_class=AliasBlockSelectable,
+            selectable=get_selectable_blocks(),
+            override=True
+        )
+    )
+
+
+class ContentPageValidator(validator.SimpleFieldValidator):
+    """Don't allow sl pages containing another AliasBlock"""
+
+    def validate(self, value):
+        """Validate international phone number on input"""
```

Comment:
!! ?? :-)

Reply:
I just copied that part. XD I'll change it 2.x too